packages/hyperion-autologging/src/ALSurfaceMutationPublisher.ts

```diff
@@ -62,10 +64,84 @@ export type InitOptions = Types.Options<
   {
     channel: ALSurfaceMutationChannel;
     cacheElementReactInfo: boolean;
+    enableDefaults?: boolean
   }
 >;
 
+function publishSurfaceInputDefaults(options: InitOptions): void {
+  const { channel } = options;
+
+  channel.addListener('al_surface_mutation_event', eventData => {
+    const { event, surface, element } = eventData;
+    if (event === 'mount_component' && surface != null && element != null) {
+      // The following is expensive, so try to jam everything we are interested in to the selector
```

Comment:
should we check capability here as well and only do this for non-impression surfaces?

Reply:
Hmm, that's a very good question. I think it does make sense to only track this for interactable surfaces, essentially whatever would have been recorded when user clicks on element. 
I'll make that change.

Reply:
Actually, this code already filters non-interactive surfaces because for each item it looks up its `parentSurfaceNode` in the dom and it can only see interactable surfaces that way. So, the impression only surface will be automatically filtered along the way.

Reply:
Ah yes good call